### PR TITLE
Automatically generated tests

### PR DESCRIPTION
**Pull request by mamaar**
The upstream Yoga project ships with various language bindings and automatically generates corresponding unit tests. Here's a generator for yoga-rs using their tool. 

This PR also updates Yoga to 1.10.0

### CODE REVIEW

**Comment by bschwind** [comment]
Hey, thanks for the PR! This will be a great addition, the manual tests were definitely not very fun.

This will take me some time to review, but I took a quick skim through it. Two points I'd like to add:

* Could you make the whitespace in the JS file consistent? The rust code in this repo uses tabs but 4 spaces is fine as well, as long as it's consistent
* Could you add a small blurb in the readme about how to generate the tests?

Perhaps we could add the test generation to `build.rs`, though I'm not sure if we'd want that or if `cargo test` would play nice with that...

**Comment by mamaar** [comment]
I will look at your points later today or tomorrow!

**Comment by mamaar** [comment]
This PR also adds Yoga as a submodule now. The version is defined in .gitmodules. Initialization and updating commands for the submodule repository are done in builder.rs, so no manual work required.

Building and testing the project is still as simple as their respective cargo command. Updating the test cases is done by running gentest/gentest.rb and should only be required when updating the upstream Yoga version.

File: README.md [hunk 1]
```diff
@@ -60,6 +60,19 @@ fn main() {
...
 ```
 
+Testing
+-------
+The unit tests are automatically generated based on upstream fixtures and should not be edited manually.
+
+    $ cargo test
+
+
+To generate the test cases, install `ChromeDriver` and the following Ruby Gems: `watir`, `watir-webdriver`. Run:
```
Comment (bschwind) [inline README.md]: Could you include a link to this page?

http://chromedriver.chromium.org

Comment (bschwind) [reply README.md]: Also a simple blurb about installing these tools would be helpful.

* Download the `chromedriver` binary and put it somewhere on your $PATH
* Linux/MacOS Example:
```bash
$ cp chromedriver /usr/local/bin
```
* Install the Ruby Gems:

```bash
$ sudo gem install watir watir-webdriver
```

* Generate the tests

```bash
$ ruby gentest/gentest.rb
```

**Comment by bschwind** [comment]
> I guess the tests changed because I ran a cargo fmt command

Ah okay, that makes sense. I guess I'd say let's leave the auto-generated tests as-is for now and let's deal with `cargo fmt` a bit later. I actually don't use `cargo fmt` at the moment and from what I can tell, isn't something that ships with stable rust yet (I might be wrong on this point)

Regarding the missing tests, if they still run correctly I'd definitely like to keep them in. If they don't run correctly, I'd like to investigate why that is. Those tests are pretty valuable at catching any mistakes in the code.

File: README.md [hunk 2]
```diff
@@ -60,6 +60,19 @@ fn main() {
...
+
+    $ ruby gentest/gentest.rb
+
+
+
 Yoga Version
```
[the remark this answers is not in the document]
Comment (mamaar) [reply README.md]: Not sure that I understand what you mean. Should I remove the text about how to generate tests?

**Comment by bschwind** [comment]
Sorry to be annoying on this one point, but isn't `watir-webdriver` required as well?

So the full install line should be:

```
sudo gem install watir watir-webdriver
```

instead of

```
sudo gem install watir
```

File: README.md [hunk 3]
```diff
@@ -60,7 +60,29 @@ fn main() {
...
+
+```
+$ cp chromedriver /usr/local/bin
+```
 
-[c5f826de8306e5fbe5963f944c75add827e096c3](https://github.com/facebook/yoga/tree/c5f826de8306e5fbe5963f944c75add827e096c3/yoga)
+Install the required Ruby gem.
+
+```
+$ gem install watir
```
[the remark this answers is not in the document]
Comment (mamaar) [reply README.md]: Installing `watir-webdriver` yields 
> With the release of Watir 6.0, the watir-webdriver gem has changed its name
to watir. Update your dependencies to use "watir", "~> 6.0"

Omitting it works for me.

[the remark this answers is not in the document]
Comment (bschwind) [reply README.md]: Ah okay, if omitting it works, then ignore the comment I just made 👌

**Comment by bschwind** [comment]
Merged! Thanks so much for your work on this, it's a very helpful change.

**Pull request merged**